scheduler/src/state_manager/mod.rs

```diff
@@ -172,14 +169,14 @@ impl StateManager {
 
     async fn update_state(&mut self) {
         let ready_workers = self.get_workers_ready().await;
-        if ready_workers.len() == 0 {
+        if ready_workers.is_empty() {
             info!("State isn't updated as there is no worker available");
             return;
         }
 
         let mut workers = ready_workers.iter().cycle();
         // Scheduling of new instances
-        for (id, workload) in self.state.iter_mut() {
+        for (_id, workload) in self.state.iter_mut() {
```

Comment:
Even better with only an underscore :D
```suggestion
        for (_, workload) in self.state.iter_mut() {
```